cura/Settings/MachineManager.py

```diff
@@ -354,11 +354,16 @@ def setActiveMachine(self, stack_id: str) -> None:
         if containers:
             Application.getInstance().setGlobalContainerStack(containers[0])
 
+        self.activeQualityChanged.emit()
```

Comment:
A better fix is to use __onInstanceContainersChanged(); That will do all this and check if there are errors (or just trigger the _onInstanceContainerChanged with any container.

Reply:
Thanks, I don't realize about that. Already changed.